internal/resource/statefulset.go

```diff
@@ -258,6 +259,16 @@ func patchPodSpec(podSpec, podSpecOverride *corev1.PodSpec) (corev1.PodSpec, err
 		sortVolumeMounts(patchedPodSpec.Containers[0].VolumeMounts)
 	}
 
+	if podSpecOverride.SecurityContext != nil && reflect.DeepEqual(*podSpecOverride.SecurityContext, corev1.PodSecurityContext{}) {
```

Comment:
I would suggest adding comments here with context about why we are handing securityContext as a special case. I can see myself getting confused in the future.